c/driver/postgresql/copy/writer.h

```diff
@@ -601,10 +667,46 @@ static inline ArrowErrorCode MakeCopyFieldWriter(
         case NANOARROW_TYPE_LARGE_BINARY:
         case NANOARROW_TYPE_LARGE_STRING:
           *out = std::make_unique<PostgresCopyBinaryDictFieldWriter>();
+          out->get()->Init(array_view);
           return NANOARROW_OK;
         default:
           break;
       }
+      break;
+    }
+    case NANOARROW_TYPE_LIST: {
+      // For now our implementation only supports primitive children types
+      // See PostgresCopyListFieldWriter::Write for limtiations
+      struct ArrowSchemaView child_schema_view;
+      NANOARROW_RETURN_NOT_OK(
+          ArrowSchemaViewInit(&child_schema_view, schema->children[0], error));
+      switch (child_schema_view.type) {
+        case NANOARROW_TYPE_INT32: {
+          // TODO: likely need to make type_resolver available here
```

Comment:
Hinted at in current comment, but I _think_ we need a way to forward the resolver or connection to this part of the code to get this to work